packages/mgt-components/src/components/mgt-file-list/mgt-file-upload/mgt-file-upload.ts

```diff
@@ -13,21 +13,60 @@ import { getSvg, SvgIcon } from '../../../utils/SvgHelper';
 import { formatBytes } from '../../../utils/Utils';
 import { IGraph, MgtBaseComponent, mgtHtml, customElement } from '@microsoft/mgt-element';
 import { ViewType } from '../../../graph/types';
-import { DriveItem } from '@microsoft/microsoft-graph-types';
+import { DriveItem, UploadSession } from '@microsoft/microsoft-graph-types';
 import {
   clearFilesCache,
   getGraphfile,
   getUploadSession,
   sendFileContent,
-  sendFileChunck,
-  deleteSessionFile
+  sendFileChunk,
+  deleteSessionFile,
+  isUploadSession
 } from '../../../graph/graph.files';
 
 import { registerFluentComponents } from '../../../utils/FluentComponents';
 import { fluentButton, fluentCheckbox, fluentProgress, fluentCard } from '@fluentui/web-components';
 
 registerFluentComponents(fluentProgress, fluentButton, fluentCheckbox, fluentCard);
 
+/**
+ * Simple union type for file system entry and directory entry types
+ */
+type FileEntry = (FileSystemDirectoryEntry | FileSystemFileEntry | FileSystemEntry) & {
```

Comment:
There are a host of different types for files/directories etc and we need to be able to use them all in our flows, so I added some type aliases and type guards to make it easy to tell which type we had to allow typesafe property access and minimize casting